doc/porting_guide.md

```diff
@@ -3,268 +3,284 @@
 
 The client provides this functionality while remaining small enough to reside on low resource MCUs, consumes minimal memory, processor power and energy consumption.
 
 The codebase is built against C99 standards.
 
-For more information place see the Xively C Client user guide in the Xively Developer Center.
+For more information please see the Xively C Client User Guide: `doc/user_guide.md`.
 
-#### Python Client
-
-The Python Client is meant for platforms where Python interpreter is available. Typically
-where there's no need for low resource consumption and where battery power or energy consumption is not a bottleneck.
-
-Python 2.7 and 3.4+ are both supported.
-
-The development version of the Xively Python Client is in this repository under xi_client_py.  A release version of this same source-base is accessible in two ways:
-
-- pip install xiPy OR
-- public git repository: [xiPy](https://github.com/xively/xipy)
 
 ## Supported Platforms
 
 The Xively Client has been deployed on many different devices already shipping in the IoT space running the following platform stacks:
 
-  - POSIX
-  - Microchip
+  - POSIX systems (OSX, Linux)
+  - Microchip TCP SDK
   - WMSDK
   - Texas Instruments CC3200
+  - STM3241G-EVAL
 
-Platforms above are officially supported and tested.
-
-Porting the Xively C Client to new platforms is accelerated by the Board Support Package (BSP). The BSP implements a series of abstract platform dependencies abstracts the platform specific implementations required by the of Xively C Client logic to a well separated space to ease the focus on porting.
+Porting the Xively C Client to new platforms is accelerated by the Board Support Package (BSP). 
 
 ## Board Support Package (BSP)
 
-The Board Support Package (BSP) is the set of functions that must be tailored to a platform's specific secure networking, memory, random, and time implementations. All of these platform-specific function calls are channeled into a few files that reside under a single directory. It is this directory (xi\_client\_c/src/bsp) that you may focus your attention when compiling the client for your particular device or platform.
+The Board Support Package (BSP) is the well-defined set of functions that the Xively Client invokes to interact with a platform's specific networking, [TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security), memory, random number generator, and time implementations. 
+
+We've organized the sources of the Xivey Client so that your BSP implementation is in just a few files that reside in the directory (`src/bsp`).  This is where you should focus your attention when writing porting the Xively Client to your device SDK.
 
 The BSP was designed to minimize the time it takes to make a port of the library. With it an engineer can ignore the MQTT codec or the non-blocking / asynchronous engine that resides in the main library source.
 
 All of the BSP **function declarations** can be found under the _xi\_client\_c/include/bsp_
```

Comment:
sorry xi_client_c still resides here. because of my escape characters. You can remove those escape chars, it is only needed by my local chrome-md-plugin. sorry.